server/src/logic/expense_operations.rs

```diff
@@ -0,0 +1,76 @@
+use super::error::ExpenseError;
+
+use crate::routes::dto::expenses::NewExpenseRequest;
+
+use chrono::NaiveDate;
+use entity::currency_type::{self, Entity as CurrencyType};
+use entity::expense::{self, Entity as Expense};
+use entity::predefined_expense::{self, Entity as PredefinedExpense};
+use entity::recurrence_type::{self, Entity as RecurrenceType};
+
+use sea_orm::ActiveValue::NotSet;
+use sea_orm::{ActiveModelTrait, ColumnTrait, DatabaseConnection, EntityTrait, QueryFilter, Set};
+
+pub async fn find_expenses_by_user_id(
+    conn: &DatabaseConnection,
+    user_id: i64,
+) -> Result<Vec<expense::Model>, ExpenseError> {
+    let expenses = Expense::find()
```

Comment:
`.all(db).await?` returns with an Vec<expense::Model> instead of Option. It makes sense, as it can simply return an empty vector if founds no result.
I would keep it as is if you don't have another idea.

Reply:
Oh my bad, I misread the code a bit. But still, is it a good practice to throw an error when there is simple no expenses? I do not think that from a design standpoint having no expenses is a reason to throw errors. On the frontend wouldnt it look strange that instead of an empty array there is an error code that needs to be handled?

Reply:
Well, this is sad. Of course, we just need an empty array if it is empty.

I was blindly following some pattern and forgot to interpret what my code does.
Fixed in: https://github.com/szattila98/veryrezsi/pull/40/commits/89f97693c574447f99a4b5ad9233b21ea63b2cc2